library/control/src/modules/ProductFeatures.rb

```diff
@@ -318,6 +318,25 @@ def Import(import_settings)
       nil
     end
 
+    # Overlay, or override, some features.
+    # The intended use is to use PushOverlay to apply some features
+    # specified by a system role (FATE#317481).
+    # PopOverlay would be called only when coming Back
```

Comment:
Sounds a bit like overengineering for me. Why not when going back simply reconstruct original features? Something like InitFeatures(true)